packages/compass-connections/src/stores/connections-store.ts

```diff
@@ -436,25 +422,50 @@ export function useConnections({
   useEffect(() => {
     // Load connections after first render.
     void loadConnections(dispatch, connectionRepository, { persistOIDCTokens });
+    void connectWithAutoConnectInfoIfAvailable();
+    async function connectWithAutoConnectInfoIfAvailable() {
+      let connectionInfo: ConnectionInfo | undefined;
+      try {
+        connectionInfo =
+          typeof getAutoConnectInfo === 'function'
+            ? await getAutoConnectInfo()
+            : undefined;
+        if (connectionInfo) {
+          log.info(
+            mongoLogId(1_001_000_160),
+            'Connection Store',
+            'Performing automatic connection attempt'
+          );
+          dispatch({
+            type: 'set-active-connection',
+            connectionInfo,
+          });
+          void connect(connectionInfo, false);
+        }
+      } catch (error) {
+        if (connectionInfo) {
+          onConnectionFailed(connectionInfo, error as Error);
+        }
```

Comment:
This looks odd: so if getting autoconnect info failed, we dispatch `connection-attempt-errored` to the internal store here, but not communicate it back to the parent? Is this expected?